### ISSUE TITLE
Hint to the "SSH Container Passthrough" / "Installation with Docker" docs

### ISSUE THREAD

**Issue by eheidt** [go-gitea/gitea#12625]
- Gitea version (or commit ref):
  - latest
- Git version:
- Operating system:
  - ubuntu 20.04 docker
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

We followed the [Installation with Docker](https://docs.gitea.io/en-us/install-with-docker/), but afterwards cloning via SSH did not work.
After some investigation, we found out that the "soft link" to authorized_keys was the problem. Making it as "hard link" solved the problem at first - we have not figured out why. After some time, we run into same problem, when somebody has added a new ssh key in his account.
After a second investigation we found out, that gitea is doing backups on the autorizid_keys - so the "hard link" was pointing to an old backup.

The final solution is mounting the git users ".ssh" directory directly into the gitea docker-container, with the additional entry under volumes: `- /home/git/.ssh/:/data/git/.ssh/`
Maybe it should be mentioned in https://docs.gitea.io/en-us/install-with-docker/#ssh-container-passthrough

````
services:
  server:
    image: gitea/gitea:latest
    environment:
      - USER_UID=1000
      - USER_GID=1000
+      - DB_TYPE=postgres
+      - DB_HOST=db:5432
+      - DB_NAME=gitea
+      - DB_USER=gitea
+      - DB_PASSWD=gitea
    restart: always
    networks:
      - gitea
    volumes:
      - ./gitea:/data
      - /home/git/.ssh/:/data/git/.ssh/
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
    ports:
      - "3000:3000"
      - "222:22"
````

**Comment by NHellFire** [comment]
I just figured out yesterday that symlinking authorized_keys wasn't working (permissions on parent directories) Another alternative to mounting .ssh in the countainer is to set the AuthorizedKeysFile to the full path just for the git user. With openssh, that's:
```bash
$ cat /etc/ssh/sshd_config.d/zgitea.conf 
Match User git
	PasswordAuthentication no
	AuthorizedKeysFile /srv/docker/gitea/data/git/.ssh/authorized_keys
```

**Comment by zeripath** [comment]
You can turn off the authorized keys backup and probably should:

```
[ssh]
SSH_BACKUP_AUTHORIZED_KEYS=false
```

I'll probably put a PR in to switch off the back up as as far as I can see it will happily make an unlimited number of backups...

**Comment by zeripath** [comment]
Would you be able to make a documentation PR for those changes?

**Comment by divbhasin** [comment]
@zeripath I can work on this. To clarify, do I need to update the docs to include OP's workaround (maybe with a note saying why)?

**Comment by zeripath** [comment]
Closed by #13030
